src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,170 @@
+package duke;
+
+import duke.myExceptions.InvalidInput;
+import duke.myExceptions.NoContent;
+import duke.myExceptions.NoTime;
+import duke.myTasks.Deadline;
+import duke.myTasks.Event;
+import duke.myTasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+//built JAR file
+public class Duke {
+
+    public static ArrayList<Todo> list = new ArrayList<>();
+    public static int index = 0;
+    public static boolean isRunning = true;
+    public static String filePath = "savefile.txt";
+
+    public static void addToList(Todo task) {
+        list.add(task);
+        index++;
+    }
+
+    public static void printList() {
+        for (int i = 0; i < index; i++) {
+            System.out.println((i + 1) + ". " + list.get(i).toString());
+        }
+    }
+
+    public static void printAddMessage(Todo task) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task.toString());
+        System.out.println("Now you have " + index + " tasks in the list.");
+    }
+
+    public static void executeInstruction(String command, String content) throws InvalidInput, NoContent, NoTime, NullPointerException, NumberFormatException, ArrayIndexOutOfBoundsException{
+        switch (command) {
+        case "bye":
+            isRunning = false;
+            save();
+            break;
+        case "list":
+            printList();
+            break;
+        case "done":
+            list.get(Integer.parseInt(content) - 1).markDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(list.get(Integer.parseInt(content) - 1).toString());
+            break;
+        case "todo":
+            Todo todo = new Todo(content);
+            addToList(todo);
+            printAddMessage(todo);
+            break;
+        case "event":
+            Event event = new Event(content);
+            addToList(event);
+            printAddMessage(event);
+            break;
+        case "deadline":
+            Deadline deadline = new Deadline(content);
+            addToList(deadline);
+            printAddMessage(deadline);
+            break;
+        default:
+            throw new InvalidInput();
+        }
+    }
+
+    public static void checkSave() {
+        Path path = Paths.get(filePath); //creates Path instance
+        try {
+            Path p = Files.createFile(path);     //creates file at specified location
+
+        } catch (IOException e) {
+            loadSave();
+        }
+    }
+
+    public static void loadSave() {
+        File f = new File(filePath);
+        try {Scanner s = new Scanner(f); // create a Scanner using the File as the source
```

Comment:
Good job in starting a try catch block is here. However, you can consider ensuring layout is consistent like in line 113.